foundation-extensions/mobile-core/platform-sercvies/network-service.md

```diff
@@ -0,0 +1,232 @@
+		null, 10000, 10000, new NetworkService.Callback() {
+			@Override
+			public void call(NetworkService.HttpConnection connection) {
+				// handle `httpConnection`
+			}
+});
+```
+
+{% endtab %}
+
+{% tab title="iOS(AEP 3.x)" %}
+
+```swift
+// Create your `NetworkRequest`, for more details see `NetworkRequest.swift`
+let networkRequest = NetworkRequest(url: url, httpMethod: .get, httpHeaders: headers)
+
+// Get an instance of the current network service
+let networkService = ServiceProvider.shared.networkService
+
+// Make a request
+networkService.connectAsync(networkRequest: networkRequest) { httpConnection in
+  // handle `httpConnection`
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+## Public Interfaces
+
+{% tabs %}
+{% tab title="Android" %}
+
+- **Networking**
+
+```java
+public interface Networking {
+	/**
+	 * Initiates an asynchronous network connection
+	 * @param request {@link NetworkRequest} used for connection
+	 * @param callback {@link NetworkCallback} that will receive the {@link HttpConnecting} instance after the connection has been made
+	 */
+	void connectAsync(final NetworkRequest request, final NetworkCallback callback);
+}
+```
+
+- **NetworkRequest**
+
+​       NetworkRequest class contains the data needed to initiate a network request. 
+
+```java
+public class NetworkRequest {
+	private final String url;
+	private final HttpMethod method;
+	private final byte[] body;
+	private final Map<String, String> headers;
+	private final int connectTimeout;
+	private final int readTimeout;
+
+	/**
+	 * Constructor for NetworkRequest
+	 * @param url {@link String} containing the full url for connection
+	 * @param method {@link com.adobe.marketing.mobile.services.HttpMethod}, for example "POST", "GET" etc.
+	 * @param body {@code byte[]} array specifying payload to send to the server
+	 * @param headers {@code Map<String, String>} containing any additional key value pairs to be used while requesting a
+	 *                        connection to the url depending on the {@code method} used
+	 * @param connectTimeout {@code int} indicating connect timeout value in seconds
+	 * @param readTimeout {@code int} indicating the timeout, in seconds, that will be used to wait for a read to finish after a successful connect
+	 */
+	public NetworkRequest(final String url,
+						  final HttpMethod method,
+						  final byte[] body,
+						  final Map<String, String> headers,
+						  final int connectTimeout,
+						  final int readTimeout) {
+		// ...
+	}
+
+	public String getUrl() {
+		return url;
+	}
+
+	public HttpMethod getMethod() {
+		return method;
+	}
+
+	public byte[] getBody() {
+		return body;
+	}
+
+	public Map<String, String> getHeaders() {
+		return headers;
+	}
+
+	public int getConnectTimeout() {
+		return connectTimeout;
+	}
+
+	public int getReadTimeout() {
+		return readTimeout;
+	}
+}
+```
+
+{% endtab %}
+
+{% tab title="iOS(AEP 3.x)" %}
+
+- **Networking**
+
+```swift
+public protocol Networking {
```

Comment:
I like how we have this split into subpages now. But i still think adding the entire protocol or class is overkill and might be too much. I am still not sure what the best approach is to document our public classes but i don't think copy pasting the entire class / protocol is best.

I think rather than have it the same style as the code. Just add the API like we do for our public APIs with a description above it like we do for our public APIs. What do you think? So instead of 

    /// Initiates an asynchronous network connection to the specified NetworkRequest.url. This API uses `URLRequest.CachePolicy.reloadIgnoringLocalCache`.
    /// - Parameters:
    ///   - networkRequest: the `NetworkRequest` used for this connection
    ///   - completionHandler:Optional completion handler which is called once the `HttpConnection` is available; it can be called from an `HttpConnectionPerformer` if `NetworkServiceOverrider` is enabled.
    ///   In case of a network error, timeout or an unexpected error, the `HttpConnection` is nil
    func connectAsync(networkRequest: NetworkRequest, completionHandler: ((HttpConnection) -> Void)?)
}


You do something like:

```
**connectAsync**

Initiates an asynchronous network connection to the specified NetworkRequest.url. This API uses URLRequest.CachePolicy.reloadIgnoringLocalCache.

**Syntax**

```swift
func connectAsync(networkRequest: NetworkRequest, completionHandler: ((HttpConnection) -> Void)?)
```

Reply:
@cdhoffmann I think it's a good idea to only list the public APIs with sample code and API syntax. For `public classes` section, we can only document the `NetworkRequest ` and `HttpConnection ` classes. Thoughts?

Reply:
That works for me!